### ISSUE TITLE
Incorrect error message on assert!() with generic methods

### ISSUE THREAD

**Issue by jiangzhe** [rust-analyzer/rust-analyzer#7762]
```rust
struct A {}

impl A {
    pub fn lookup<K, V>(&self, k: &K) -> Option<V> {
        None
    }
}

fn main() {
    let a = A {};
    a.lookup::<_, u64>(&0u32).is_none();
    assert!(a.lookup::<_, u64>(&0u32).is_none());
}
```

I use vs-code 1.53.2 with rust-analyzer 0.2.497.
The program above compiles and runs fine.
Second line in main() function is ok.
Third line reports `mismatched-arg-count` error message, when adding assert!() macro to the generic method.

**Comment by lnicola** [comment]
This only happens with generics.

**Comment by flodiebold** [comment]
This looks like it could be the same issue as #7640; checking the rust-analyzer macro expansion would be good to confirm.
